lib/libc/minimal/source/stdlib/strtoul.c

```diff
@@ -76,9 +76,9 @@ unsigned long strtoul(const char *nptr, char **endptr, register int base)
 	cutlim = (unsigned long)ULONG_MAX % (unsigned long)base;
 	for (acc = 0, any = 0;; c = *s++) {
 		if (isdigit((unsigned char)c) != 0) {
-			c -= '0';
+			c = (char)c - '0';
 		} else if (isalpha((unsigned char)c) != 0) {
-			c -= isupper((unsigned char)c) != 0 ? 'A' - 10 : 'a' - 10;
+			c = (char)c - (isupper(c) != 0 ? 'A' : 'a') + 10;
```

Comment:
The (unsigned char) cast is required for the parameter to isupper as that function is only defined on unsigned char and EOF values. I believe a reasonable alternative would be to insert the cast into the fetch in line 77 == `c = (unsigned char) *s++`.